src/mono/dlls/dbgshim/dbgshim.cpp

```diff
@@ -70,8 +108,18 @@ CreateProcessForLaunch(
     ZeroMemory(&startupInfo, sizeof(startupInfo));
 
     startupInfo.cb = sizeof(startupInfo);
-
-    putenv("MONO_ENV_OPTIONS='--debugger-agent=transport=dt_socket,address=127.0.0.1:pid_based,server=n,suspend=y,loglevel=10,timeout=100000'");
+    char* c_android_adb_path = getenv("ANDROID_ADB_PATH");
+
+    if (strlen(c_android_adb_path) > 0) {
+        HRESULT ret = RunAndroidCmd(c_android_adb_path, W("shell setprop debug.mono.extra \"debug=10.0.2.2:56000,loglevel=10,timeout=100000000000000\""));
```

Comment:
this is ok for now, but eventually we should call out here to something in the workload (or in the runtime pack, or... somewhere) that will run the commands for us.  We don't want to put it into dbgshim - we should put it some place that is updated in sync with the workloads, not with the debugger.

Reply:
Okay, I will talk to Android guys.